Makefile.toml

```diff
@@ -127,13 +127,30 @@ AMI_DATA_FILE_SUFFIX = "amis.json"
 #    5: a final `quick` test on the downgraded instances
 # TESTSYS_STARTING_IMAGE_ID can be used to provide the correct starting image for migration tests.
 TESTSYS_TEST = "quick"
-# The path to the testsys cluster's kubeconfig file. This is used for all testsys calls.
-TESTSYS_KUBECONFIG_PATH = "${BUILDSYS_ROOT_DIR}/testsys.kubeconfig"
+# The default path to the testsys cluster's kubeconfig file. This is used for all testsys calls.
+CARGO_MAKE_DEFAULT_TESTSYS_KUBECONFIG_PATH = "${BUILDSYS_ROOT_DIR}/testsys.kubeconfig"
 # The last released version of bottlerocket.
 TESTSYS_STARTING_VERSION = { script = ["git tag --list --sort=version:refname 'v*' | tail -1"] }
 # The commit for the last release of bottlerocket.
 TESTSYS_STARTING_COMMIT = { script = ["git describe --tag ${TESTSYS_STARTING_VERSION} --always --exclude '*' || echo 00000000"] }
 
+# Determines the kubeconfig that should be used by testsys. If no kubeconfig was provided and the
+# default kubeconfig location does not exist, use the users default kubeconfig.
+CARGO_MAKE_TESTSYS_KUBECONFIG_ARG = {script = [
+   '''
+      if [[ -n ${TESTSYS_KUBECONFIG_PATH} ]]; then
+         # If the user provides a kubeconfig path it should be used.
+         echo "--kubeconfig ${TESTSYS_KUBECONFIG_PATH}"
+      elif [[ -f "${CARGO_MAKE_DEFAULT_TESTSYS_KUBECONFIG_PATH}" ]]; then
+         # If the default kubeconfig exists it should be used.
+         echo "--kubeconfig ${CARGO_MAKE_DEFAULT_TESTSYS_KUBECONFIG_PATH}"
+      fi
+   '''
+]}
```

Comment:
Is there a need for both `TESTSYS_KUBECONFIG_PATH` and `CARGO_MAKE_DEFAULT_TESTSYS_KUBECONFIG_PATH` ?

Couldn't we just keep the old default value and then they could override the kubeconfig by setting that one variable via `-e TESTSYS_KUBECONFIG_PATH`?

If you tried that and it didn't work, it might help to move this into the `env.development` block. I believe dependencies between env variables was a feature added in a very recent `cargo-make` release, but I don't quite want to rely on that yet.

Reply:
Do we need to know that? It seems like it would be nice if this worked like some of our other overrides (e.g. `-e BUILDSYS_VARIANT`) where if not specified it just uses the default, otherwise the user can override the variable with their custom location.

Reply:
The main issue is, if the user supplies a kubeconfig, but there is a typo and it doesn't exist, it would default to using an unplanned kubeconfig which should not happen.